### ISSUE TITLE
Button component: aria-disabled focusable button should use consistent focus style

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#62278]
### Description

Discovered while working on https://github.com/WordPress/gutenberg/pull/62267

Buttons can be made `aria-disabled` to still be focusable and this pattern has been recently made more solid in https://github.com/WordPress/gutenberg/pull/62080

When `aria-disabled` buttons receive focus, their focus style should be consistent and this should be managed in the Button component itself. Instead, I observed that sometimes some `aria-disabled` buttons use a custom focus style.

An important note for accessibility is that:
- Disabled controls don't have to meet the color contrast requirement.
- However, the entire purpose of making `aria-disabled` buttons still focusable is to make them discoverable and perceivable when they receive focus. This should apply also to the focus style so that I'd argue the focus style should meet the contrast requirement. Feedback and thoughts welcome Cc @joedolson 

The current situation of the focus style for `aria-disabled` buttons is pretty inconsistent and should be improved.

1
The Default button uses [some opacity](https://github.com/WordPress/gutenberg/blob/e57eed571e8aaa7d65394bdcf861b103dee9843f/packages/components/src/button/style.scss#L246-L250) so that both the text and the focus outline are lighter:

![default](https://github.com/WordPress/gutenberg/assets/1682452/fb758dc1-d4a1-41e6-b6b8-b5f001e2c922)

2
The Primary button only makes the text lighter. The outline (a box-shadow actually) is different:

![primary](https://github.com/WordPress/gutenberg/assets/1682452/e3cb1543-21eb-48a3-a25b-7718a9dcadc2)

3
The Secondary button only makes the text color lighter. Anyways, the text color is different from the one of the Default button (which uses opacity):

![secondary](https://github.com/WordPress/gutenberg/assets/1682452/90474352-abcc-4cc1-82af-81132d5def00)

4
Same for the Tertiary button:

![tertiary](https://github.com/WordPress/gutenberg/assets/1682452/d18b37ad-6d27-4ad1-b7dd-ab562a825404)

### Step-by-step reproduction instructions

- Go to the Storybook https://wordpress.github.io/gutenberg/?path=/docs/components-button--docs
- Test the 4 Button variants: Default, Primary, Secondary, and Tertiary.
- Set both the `__experimentalIsFocusable` and `disabled` props to `true`.
- Move focus to the Button preview at the top of the page.
- Observe the focus style inconsistencies.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by jasmussen** [comment]
What design feedback are you looking for? I'd echo this seems like a bug as far as the appearance of the focus style on the disabled button.

**Comment by afercia** [comment]
From a design perspective I would like to know what is the desired focus style for aria-disabled buttons.

This one, where the blue outline is dimmed:

![Screenshot 2024-06-10 at 14 20 17](https://github.com/WordPress/gutenberg/assets/1682452/d0c03221-cc81-4047-a98d-6fc007c01e8f)

or this one where the blue outline is the standard one?

![Screenshot 2024-06-10 at 14 25 18](https://github.com/WordPress/gutenberg/assets/1682452/4f079575-bda3-4509-94dd-be07771e5dd0)

In this case, when the 'Show button text labels' preference is enabled, the button shows text and also becomes `tertiary` thus surfacing the focus style inconsistency.

From an accessibility perspective, I'm not sure there is a standard for this case. However, I'd tend to think the focus style should be clearly visible for aria-disabled controls. Cc @joedolson any thoughts?

**Comment by joedolson** [comment]
The focus needs to be fully visible. While it is acceptable to have disabled components below color contrast, per WCAG 1.4.3, that only refers to the component itself. The focus outline is a positional marker to allow a user to identify their current position in the page, and that needs to meet contrast guidelines. The user's need to know where they are in the interface doesn't go away just because the control is disabled.

**Comment by afercia** [comment]
One more case to adjust: when hovered, the text of the `is-destructive` button becomes gray. Is thould stay light red.

![Screenshot 2024-06-11 at 11 19 06](https://github.com/WordPress/gutenberg/assets/1682452/f2b13024-b059-4525-a1a9-802c35d54b93)
